test/fixtures/08-tasks/task-handlers.ts

```diff
@@ -0,0 +1,62 @@
+import type { FromSchema } from 'json-schema-to-ts';
+import { sbvrUtils } from '../../../src/server-glue/module';
+
+// Define JSON schema for accepted parameters
+const createDeviceParamsSchema = {
+	type: 'object',
+	properties: {
+		name: {
+			type: 'string',
+		},
+		type: {
+			type: 'string',
+		},
+	},
+	required: ['name', 'type'],
+	additionalProperties: false,
+} as const;
+
+// Generate type from schema and export for callers to use
+export type CreateDeviceParams = FromSchema<typeof createDeviceParamsSchema>;
+
+export const initTaskHandlers = () => {
+	sbvrUtils.addTaskHandler(
+		'create_device',
+		async (options) => {
+			try {
+				const params = options.params as CreateDeviceParams;
+				await options.api.post({
+					apiPrefix: '/example/',
+					resource: 'device',
+					passthrough: {
+						tx: options.tx,
```

Comment:
My understanding is this tx comes from poll() (or start()?) and it always commits in the end. This means that, in general, handlers can do some work and then fail in some way but the work they did will not be rolled back. Is this intentional?

Reply:
@dfunckt Thanks for checking out the PR! Initially the decision was to expect handlers to do some of the lifting here, but I see what you mean and agree. I've pushed a (somewhat hastily put together) commit that should achieve this, and asserts as much in a test. Let me know what you think!

Reply:
Thanks @joshbwlng. This looks like it’ll work nicely. The only thing I’m not sure of is if Postgres will allow a new transaction (the handlerTx) on the same connection that already has a transaction open (the “logistics” one).

Reply:
I'm honestly not sure either, which is why I've kept it in its own commit for now. Will come back to this when I have more time.

Reply:
@dfunckt Pushed again (this time with a squash), using a savepoint instead. Each savepoint is unique to the task being executed and depending on the result we can either release the savepoint or roll back to it. Much cleaner in my opinion.

Reply:
We do currently nest transactions in our migration code [here](https://github.com/balena-io/pinejs/blob/master/src/migrator/async.ts#L236C1-L236C12), so going back to the way I had it before at least is a proper option. I switched to savepoints (a) to experiment and (b) because we don't use the transaction for "logistics" before passing it off to the handler or executing savepoint release/rollback.